jsx/App/Stories/Story/lib/txt_sync.js

```diff
@@ -11,67 +36,104 @@ export function setupTextSync() {
+                changeSentenceColorToHighlight(i);
+            } else {
+                changeSentenceColorToNormal(i);
+            }
+        }
+        updateTimestampQuery(sentId);
     }
-    // }
 
     // I/P: t, an integer number of milliseconds
     // O/P: the player updates to the given time
     // Status: untested
-    function jumpToTime(t) {
-        updateTimestampQuery(t);
-        try {
-            const media = $("[data-live='true']").get(0);
+    // This function adjusts the AV file(s)' timestamp according to the 
+    // selected sentence's URL
+    function setMediaCurrentTime(t) {
+        // Change by Holly, October 3, 2020: I moved the updateTimestampQuery call to 
+        // be part of the onclick of .timeStamp.
+        // Also, I removed the try-catch block so that no error is thrown for non-AV files. 
+        const media = $("[data-live='true']").get(0);
+        if (media) {
             media.currentTime = (t + 2) / 1000;
         }
-        catch(err) {
-            console.log(err);
-            console.log("We think you tried to jump to time before the MEDIA element was created.")
+        // try {
+        //     const media = $("[data-live='true']").get(0);
+        //     media.currentTime = (t + 2) / 1000;
+        // }
+        // catch(err) {
+        //     console.log(err);
+        //     console.log("We think you tried to jump to time before the MEDIA element was created.")
+        // }
+    }
+
+    /* Updates the URL according to a sentence's index id. */
+    function updateTimestampQuery(newSentenceIndex) {
+        if (window.history.replaceState) {
+            // const newurl = window.location.protocol + "//" + window.location.host + window.location.pathname + window.location.hash.replace(/\?.*$/, ''); // hacky...
+
+            // (Holly:) The original code had -1 in the timestamp after ? --- why is this -1 necessary? 
+            const newurl = window.location.href.replace(/\?.*$/,'') + `?${newSentenceIndex}`; // hacky...
+            window.history.replaceState({ path: newurl }, '', newurl);
         }
     }
 
-    $(".timeStamp").click(function() {
-        jumpToTime($(this).data('start_time'));
-        // document.location.search = $(this).data('start_time');
-    });
 
     // Jump to timestamp:
     // Source: http://snydersoft.com/blog/2009/11/14/get-parameters-to-html-page-with-javascript/
     $( document ).ready(function() {
+        // This sentenceTimestampId is the timestamp for a file with AV, 
+        // and a sentence id for a file without AV. 
         let query_index = document.URL.indexOf("?");
-        let startTime = Number(document.URL.substr(query_index+1));
-        if (isFinite(startTime)) {
+        let sentenceTimestampId = Number(document.URL.substr(query_index+1));
+        if (isFinite(sentenceTimestampId)) {
             // startTime = startTime + 3 // do not remove (result of hacky solution further up in this file)
```

Comment:
more commented-out code to delete

Reply:
(even though it predates your edits, we might as well clean it up)

Reply:
Updated